Install fails

Neither

`mvn install
`

nor 
```
cd duke-farm
./deploy.sh 

```
work.

This is the output:
```
./deploy.sh 
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ nl.vaneijndhoven:dukes:1.0-SNAPSHOT, /Users/wf/Documents/workspace/dukes/pom.xml, line 74, column 21
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: nl.vaneijndhoven.dukes:general-lee:jar -> duplicate declaration of version 1.0-SNAPSHOT @ nl.vaneijndhoven.dukes:boars-nest:[unknown-version], /Users/wf/Documents/workspace/dukes/boars-nest/pom.xml, line 49, column 21
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: nl.vaneijndhoven.dukes:hazard-county:jar -> duplicate declaration of version 1.0-SNAPSHOT @ nl.vaneijndhoven.dukes:boars-nest:[unknown-version], /Users/wf/Documents/workspace/dukes/boars-nest/pom.xml, line 84, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ nl.vaneijndhoven:dukes:1.0-SNAPSHOT, /Users/wf/Documents/workspace/dukes/pom.xml, line 74, column 21
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: nl.vaneijndhoven.dukes:general-lee:jar -> duplicate declaration of version 1.0-SNAPSHOT @ nl.vaneijndhoven.dukes:boss-hogg:[unknown-version], /Users/wf/Documents/workspace/dukes/boss-hogg/pom.xml, line 49, column 21
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: nl.vaneijndhoven.dukes:daisy:jar -> duplicate declaration of version 1.0-SNAPSHOT @ nl.vaneijndhoven.dukes:boss-hogg:[unknown-version], /Users/wf/Documents/workspace/dukes/boss-hogg/pom.xml, line 69, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ nl.vaneijndhoven:dukes:1.0-SNAPSHOT, /Users/wf/Documents/workspace/dukes/pom.xml, line 74, column 21
[WARNING] 'dependencies.dependency.systemPath' for ext:opencv:jar should not point at files within the project directory, ${project.basedir}/src/main/resources/lib/osx-bj/opencv-330.jar will be unresolvable by dependent projects @ nl.vaneijndhoven.dukes:cletus:[unknown-version], /Users/wf/Documents/workspace/dukes/cletus/pom.xml, line 27, column 25
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ nl.vaneijndhoven:dukes:1.0-SNAPSHOT, /Users/wf/Documents/workspace/dukes/pom.xml, line 74, column 21
[WARNING] 'dependencies.dependency.systemPath' for ext:opencv:jar refers to a non-existing file /Users/wf/Documents/workspace/dukes/cletus/src/main/resources/lib/osx-bj/opencv-330.jar @ nl.vaneijndhoven.dukes:cletus:[unknown-version], /Users/wf/Documents/workspace/dukes/cletus/pom.xml, line 27, column 25
[WARNING] 'dependencies.dependency.systemPath' for ext:opencv:jar should not point at files within the project directory, ${project.basedir}/src/main/resources/lib/osx-bj/opencv-330.jar will be unresolvable by dependent projects @ nl.vaneijndhoven.dukes:cooter:[unknown-version], /Users/wf/Documents/workspace/dukes/cooter/pom.xml, line 46, column 25
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ nl.vaneijndhoven:dukes:1.0-SNAPSHOT, /Users/wf/Documents/workspace/dukes/pom.xml, line 74, column 21
[WARNING] 'dependencies.dependency.systemPath' for ext:opencv:jar refers to a non-existing file /Users/wf/Documents/workspace/dukes/cooter/src/main/resources/lib/osx-bj/opencv-330.jar @ nl.vaneijndhoven.dukes:cooter:[unknown-version], /Users/wf/Documents/workspace/dukes/cooter/pom.xml, line 46, column 25
[WARNING] 'dependencies.dependency.systemPath' for ext:opencv:jar should not point at files within the project directory, ${project.basedir}/src/main/resources/lib/osx-bj/opencv-330.jar will be unresolvable by dependent projects @ nl.vaneijndhoven.dukes:daisy:[unknown-version], /Users/wf/Documents/workspace/dukes/daisy/pom.xml, line 42, column 25
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: nl.vaneijndhoven.dukes:cooter:jar -> duplicate declaration of version 1.0-SNAPSHOT @ nl.vaneijndhoven.dukes:daisy:[unknown-version], /Users/wf/Documents/workspace/dukes/daisy/pom.xml, line 52, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ nl.vaneijndhoven:dukes:1.0-SNAPSHOT, /Users/wf/Documents/workspace/dukes/pom.xml, line 74, column 21
[WARNING] 'dependencies.dependency.systemPath' for ext:opencv:jar refers to a non-existing file /Users/wf/Documents/workspace/dukes/daisy/src/main/resources/lib/osx-bj/opencv-330.jar @ nl.vaneijndhoven.dukes:daisy:[unknown-version], /Users/wf/Documents/workspace/dukes/daisy/pom.xml, line 42, column 25
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: nl.vaneijndhoven.dukes:general-lee:jar -> duplicate declaration of version 1.0-SNAPSHOT @ nl.vaneijndhoven.dukes:duke-farm:[unknown-version], /Users/wf/Documents/workspace/dukes/duke-farm/pom.xml, line 45, column 21
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: nl.vaneijndhoven.dukes:daisy:jar -> duplicate declaration of version 1.0-SNAPSHOT @ nl.vaneijndhoven.dukes:duke-farm:[unknown-version], /Users/wf/Documents/workspace/dukes/duke-farm/pom.xml, line 65, column 21
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: nl.vaneijndhoven.dukes:hazard-county:jar -> duplicate declaration of version 1.0-SNAPSHOT @ nl.vaneijndhoven.dukes:duke-farm:[unknown-version], /Users/wf/Documents/workspace/dukes/duke-farm/pom.xml, line 80, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ nl.vaneijndhoven:dukes:1.0-SNAPSHOT, /Users/wf/Documents/workspace/dukes/pom.xml, line 74, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ nl.vaneijndhoven:dukes:1.0-SNAPSHOT, /Users/wf/Documents/workspace/dukes/pom.xml, line 74, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ nl.vaneijndhoven:dukes:1.0-SNAPSHOT, /Users/wf/Documents/workspace/dukes/pom.xml, line 74, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ nl.vaneijndhoven:dukes:1.0-SNAPSHOT, /Users/wf/Documents/workspace/dukes/pom.xml, line 74, column 21
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: nl.vaneijndhoven.dukes:hazard-county:jar -> duplicate declaration of version 1.0-SNAPSHOT @ nl.vaneijndhoven.dukes:roscoe:[unknown-version], /Users/wf/Documents/workspace/dukes/roscoe/pom.xml, line 20, column 21
[WARNING] 'dependencies.dependency.systemPath' for ext:opencv:jar should not point at files within the project directory, ${project.basedir}/src/main/resources/lib/osx-bj/opencv-330.jar will be unresolvable by dependent projects @ nl.vaneijndhoven.dukes:roscoe:[unknown-version], /Users/wf/Documents/workspace/dukes/roscoe/pom.xml, line 30, column 25
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: io.vertx:vertx-rx-java:jar -> duplicate declaration of version (?) @ nl.vaneijndhoven.dukes:roscoe:[unknown-version], /Users/wf/Documents/workspace/dukes/roscoe/pom.xml, line 32, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ nl.vaneijndhoven:dukes:1.0-SNAPSHOT, /Users/wf/Documents/workspace/dukes/pom.xml, line 74, column 21
[WARNING] 'dependencies.dependency.systemPath' for ext:opencv:jar refers to a non-existing file /Users/wf/Documents/workspace/dukes/roscoe/src/main/resources/lib/osx-bj/opencv-330.jar @ nl.vaneijndhoven.dukes:roscoe:[unknown-version], /Users/wf/Documents/workspace/dukes/roscoe/pom.xml, line 30, column 25
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ nl.vaneijndhoven:dukes:1.0-SNAPSHOT, /Users/wf/Documents/workspace/dukes/pom.xml, line 74, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ nl.vaneijndhoven:dukes:1.0-SNAPSHOT, /Users/wf/Documents/workspace/dukes/pom.xml, line 74, column 21
[FATAL] 'dependencies.dependency. nl.vaneijndhoven.dukes:hazard-county:1.0-SNAPSHOT' for nl.vaneijndhoven.dukes:hazard-county:1.0-SNAPSHOT is referencing itself. @ nl.vaneijndhoven.dukes:hazard-county:[unknown-version], /Users/wf/Documents/workspace/dukes/hazard-county/pom.xml, line 34, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ nl.vaneijndhoven:dukes:1.0-SNAPSHOT, /Users/wf/Documents/workspace/dukes/pom.xml, line 74, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 74, column 21
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project nl.vaneijndhoven.dukes:hazard-county:1.0-SNAPSHOT (/Users/wf/Documents/workspace/dukes/hazard-county/pom.xml) has 1 error
[ERROR]     'dependencies.dependency. nl.vaneijndhoven.dukes:hazard-county:1.0-SNAPSHOT' for nl.vaneijndhoven.dukes:hazard-county:1.0-SNAPSHOT is referencing itself. @ nl.vaneijndhoven.dukes:hazard-county:[unknown-version], /Users/wf/Documents/workspace/dukes/hazard-county/pom.xml, line 34, column 21
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
ls: target/: No such file or directory

```

--- a/.gitignore
+++ b/.gitignore
@@ -46,6 +46,7 @@ Temporary Items
 .mtj.tmp/
 
 # Package Files #
+# please note this also ignores the necessary opencv.jar
 *.jar
 *.war
 *.ear

--- a/README.md
+++ b/README.md
@@ -22,6 +22,9 @@ The self driving car consists of modules with the following responsibilities:
 | Uncle Jesse   | JavaFX   | laptop  | JavaFX app to monitor the car
 
 
+Preparation
+==
+Get an opencv.jar for your platform
  
 Getting started
 ===

--- a/boars-nest/pom.xml
+++ b/boars-nest/pom.xml
@@ -40,16 +40,10 @@
         <dependency>
             <groupId>com.hazelcast</groupId>
             <artifactId>hazelcast</artifactId>
-            <version>3.3.3</version>
         </dependency>
         <dependency>
             <groupId>io.vertx</groupId>
             <artifactId>vertx-hazelcast</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>nl.vaneijndhoven.dukes</groupId>
-            <artifactId>general-lee</artifactId>
-            <version>1.0-SNAPSHOT</version>
         </dependency>
         <dependency>
             <groupId>nl.vaneijndhoven.dukes</groupId>
@@ -79,11 +73,6 @@
         <dependency>
             <groupId>nl.vaneijndhoven.dukes</groupId>
             <artifactId>uncle-jesse</artifactId>
-            <version>1.0-SNAPSHOT</version>
-        </dependency>
-        <dependency>
-            <groupId>nl.vaneijndhoven.dukes</groupId>
-            <artifactId>hazard-county</artifactId>
             <version>1.0-SNAPSHOT</version>
         </dependency>
         <dependency>

--- a/boss-hogg/pom.xml
+++ b/boss-hogg/pom.xml
@@ -45,11 +45,7 @@
         <dependency>
             <groupId>io.vertx</groupId>
             <artifactId>vertx-hazelcast</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>nl.vaneijndhoven.dukes</groupId>
-            <artifactId>general-lee</artifactId>
-            <version>1.0-SNAPSHOT</version>
+            <version>3.3.3</version>
         </dependency>
         <dependency>
             <groupId>nl.vaneijndhoven.dukes</groupId>
@@ -59,11 +55,6 @@
         <dependency>
             <groupId>nl.vaneijndhoven.dukes</groupId>
             <artifactId>bo</artifactId>
-            <version>1.0-SNAPSHOT</version>
-        </dependency>
-        <dependency>
-            <groupId>nl.vaneijndhoven.dukes</groupId>
-            <artifactId>daisy</artifactId>
             <version>1.0-SNAPSHOT</version>
         </dependency>
         <dependency>

--- a/daisy/pom.xml
+++ b/daisy/pom.xml
@@ -49,11 +49,6 @@
             <groupId>ch.qos.logback</groupId>
             <artifactId>logback-classic</artifactId>
         </dependency>
-        <dependency>
-            <groupId>nl.vaneijndhoven.dukes</groupId>
-            <artifactId>cooter</artifactId>
-            <version>1.0-SNAPSHOT</version>
-        </dependency>
     </dependencies>
 
 </project>

--- a/duke-farm/pom.xml
+++ b/duke-farm/pom.xml
@@ -44,11 +44,6 @@
         </dependency>
         <dependency>
             <groupId>nl.vaneijndhoven.dukes</groupId>
-            <artifactId>general-lee</artifactId>
-            <version>1.0-SNAPSHOT</version>
-        </dependency>
-        <dependency>
-            <groupId>nl.vaneijndhoven.dukes</groupId>
             <artifactId>flash</artifactId>
             <version>1.0-SNAPSHOT</version>
         </dependency>
@@ -64,22 +59,12 @@
         </dependency>
         <dependency>
             <groupId>nl.vaneijndhoven.dukes</groupId>
-            <artifactId>daisy</artifactId>
-            <version>1.0-SNAPSHOT</version>
-        </dependency>
-        <dependency>
-            <groupId>nl.vaneijndhoven.dukes</groupId>
             <artifactId>luke</artifactId>
             <version>1.0-SNAPSHOT</version>
         </dependency>
         <dependency>
             <groupId>nl.vaneijndhoven.dukes</groupId>
             <artifactId>uncle-jesse</artifactId>
-            <version>1.0-SNAPSHOT</version>
-        </dependency>
-        <dependency>
-            <groupId>nl.vaneijndhoven.dukes</groupId>
-            <artifactId>hazard-county</artifactId>
             <version>1.0-SNAPSHOT</version>
         </dependency>
         <dependency>

--- a/hazard-county/pom.xml
+++ b/hazard-county/pom.xml
@@ -32,14 +32,8 @@
             <artifactId>vertx-web</artifactId>
         </dependency>
         <dependency>
-            <groupId>nl.vaneijndhoven.dukes</groupId>
-            <artifactId>hazard-county</artifactId>
-            <version>1.0-SNAPSHOT</version>
-        </dependency>
-        <dependency>
             <groupId>com.hazelcast</groupId>
             <artifactId>hazelcast</artifactId>
-            <version>3.3.3</version>
         </dependency>
         <dependency>
             <groupId>io.vertx</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -11,6 +11,8 @@
 
     <properties>
         <vertx.version>3.5.4</vertx.version>
+        <hazelcast.version>3.9</hazelcast.version>
+        <logback-classic.version>1.1.5</logback-classic.version>
     </properties>
 
     <dependencies>
@@ -22,6 +24,11 @@
 
     <dependencyManagement>
         <dependencies>
+            <dependency>
+                <groupId>com.hazelcast</groupId>
+                <artifactId>hazelcast</artifactId>
+                <version>${hazelcast.version}</version>
+            </dependency>
             <dependency>
                 <groupId>io.vertx</groupId>
                 <artifactId>vertx-core</artifactId>
@@ -45,7 +52,7 @@
             <dependency>
                 <groupId>ch.qos.logback</groupId>
                 <artifactId>logback-classic</artifactId>
-                <version>1.1.5</version>
+                  <version>${logback-classic.version}</version>
             </dependency>
         </dependencies>
 
@@ -72,8 +79,10 @@
     <build>
         <plugins>
             <plugin>
+                <!-- https://mvnrepository.com/artifact/org.apache.maven.plugins/maven-compiler-plugin -->
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.1</version>
                 <configuration>
                     <source>1.8</source>
                     <target>1.8</target>

--- a/roscoe/pom.xml
+++ b/roscoe/pom.xml
@@ -3,6 +3,14 @@
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>nl.vaneijndhoven</groupId>
+        <artifactId>dukes</artifactId>
+        <version>1.0-SNAPSHOT</version>
+    </parent>
+
+    <groupId>nl.vaneijndhoven.dukes</groupId>
+    <artifactId>roscoe</artifactId>
     <dependencies>
         <dependency>
             <groupId>io.vertx</groupId>
@@ -18,11 +26,6 @@
             <artifactId>logback-classic</artifactId>
         </dependency>
         <dependency>
-            <groupId>nl.vaneijndhoven.dukes</groupId>
-            <artifactId>hazard-county</artifactId>
-            <version>1.0-SNAPSHOT</version>
-        </dependency>
-        <dependency>
             <groupId>ext</groupId>
             <artifactId>opencv</artifactId>
             <version>1.0-SNAPSHOT</version>
@@ -30,24 +33,9 @@
             <systemPath>${project.basedir}/src/main/resources/lib/osx-bj/opencv-330.jar</systemPath>
         </dependency>
         <dependency>
-            <groupId>io.vertx</groupId>
-            <artifactId>vertx-rx-java</artifactId>
-        </dependency>
-        <dependency>
             <groupId>nl.vaneijndhoven.dukes</groupId>
             <artifactId>daisy</artifactId>
             <version>1.0-SNAPSHOT</version>
         </dependency>
     </dependencies>
-
-    <parent>
-        <groupId>nl.vaneijndhoven</groupId>
-        <artifactId>dukes</artifactId>
-        <version>1.0-SNAPSHOT</version>
-    </parent>
-
-    <groupId>nl.vaneijndhoven.dukes</groupId>
-    <artifactId>roscoe</artifactId>
-
-
 </project>

